docs/aws/audit/lambdamonitoring/rules/admin_privilege.mdx

```diff
@@ -23,6 +23,272 @@ PCIDSS, HIPAA
+5. Once you have the list of policies attached to your Lambda functions, you can check each policy to see if it has administrative permissions. To do this, click on the policy name to open its details page. In the "Permissions" tab, look for permissions that include "AdministratorAccess". If such permissions are found, then the Lambda function has administrative permissions, which is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Lambda functions in your AWS account. You can do this by using the AWS CLI command `aws lambda list-functions`. This command will return a list of all the Lambda functions in your AWS account.
+
+2. Once you have the list of Lambda functions, you can check the permissions attached to each function. You can do this by using the AWS CLI command `aws lambda get-policy --function-name <function-name>`. Replace `<function-name>` with the name of the Lambda function you want to check. This command will return the policy attached to the specified Lambda function.
+
+3. Now, you need to parse the output of the `get-policy` command to check if the Lambda function has administrative permissions. You can do this by looking for the `"Effect": "Allow"` and `"Action": "lambda:*"` fields in the policy. If these fields are present, it means that the Lambda function has administrative permissions.
+
+4. Repeat steps 2 and 3 for each Lambda function in your AWS account. If any function has administrative permissions, it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+import boto3
+```
+
+2. Create a session using your AWS credentials. You can also use the AWS CLI's default configuration.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    aws_session_token='SESSION_TOKEN',
+)
+```
+
+3. Create a client for AWS Lambda and IAM:
+
+```python
+lambda_client = session.client('lambda')
+iam_client = session.client('iam')
+```
+
+4. List all Lambda functions and check their execution role:
+
+```python
+functions = lambda_client.list_functions()
+for function in functions['Functions']:
+    role_name = function['Role'].split('/')[-1]
+    policy = iam_client.get_role_policy(RoleName=role_name, PolicyName='inline_policy_for_lambda')
+    if 'Statement' in policy['PolicyDocument']:
+        for statement in policy['PolicyDocument']['Statement']:
+            if statement['Effect'] == 'Allow' and 'Action' in statement and '*' in statement['Action']:
+                print(f"Lambda function {function['FunctionName']} has administrative permissions.")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>